eval/eval.go

```diff
@@ -17,6 +17,61 @@ import (
 // See todo in token about publishing all of them.
 var unquoteToken = token.ByType(token.UNQUOTE)
 
+func (s *State) compoundAssignNested(node ast.Node, operator token.Type, value object.Object) (object.Object, *object.Error) {
+	n, ok := node.(*ast.IndexExpression)
+	if !ok {
+		err := s.NewError("assignment to non identifier: " + node.Value().DebugString())
```

Comment:
should probably be 
return s.Errorfp("assignment to non identifier:", node.Value().DebugString()) // or maybe .Inspect() not sure

(and assignNested is wrong too, not sure why I have these 
		err := s.NewError("assignment to non identifier: " + node.Value().DebugString())
maybe I wrote that before I wrote Errorfp or forgot - or you can leave it as is and I/we can refactor that seperately)

Reply:
moving to #385